api/request.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 Fraunhofer AISEC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//           $$\                           $$\ $$\   $$\
+//           $$ |                          $$ |\__|  $$ |
+//  $$$$$$$\ $$ | $$$$$$\  $$\   $$\  $$$$$$$ |$$\ $$$$$$\    $$$$$$\   $$$$$$\
+// $$  _____|$$ |$$  __$$\ $$ |  $$ |$$  __$$ |$$ |\_$$  _|  $$  __$$\ $$  __$$\
+// $$ /      $$ |$$ /  $$ |$$ |  $$ |$$ /  $$ |$$ |  $$ |    $$ /  $$ |$$ | \__|
+// $$ |      $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$\ $$ |  $$ |$$ |
+// \$$$$$$\  $$ |\$$$$$   |\$$$$$   |\$$$$$$  |$$ |  \$$$   |\$$$$$   |$$ |
+//  \_______|\__| \______/  \______/  \_______|\__|   \____/  \______/ \__|
+//
+// This file is part of Clouditor Community Edition.
+
+package api
+
+import (
+	"errors"
+	"strings"
+
+	"golang.org/x/exp/slices"
+	"google.golang.org/protobuf/proto"
+
+	"clouditor.io/clouditor/internal/util"
+)
+
+var (
+	ErrInvalidColumnName = errors.New("column name is invalid")
+	ErrRequestIsNil      = errors.New("request is empty")
+)
+
+// SortedListRequest indicates a proto message being a request for ListXXX RPCs
+type SortedListRequest interface {
+	GetOrderBy() string
+	GetAsc() bool
+	proto.Message
+	PaginatedRequest
+}
+
+func ValidateListRequest[T proto.Message](req SortedListRequest) (err error) {
+	// req must be non-nil
+	if req == nil {
+		err = ErrRequestIsNil
+		return
+	}
+
+	// Avoid DB injections by whitelisting the valid orderBy statements
```

Comment:
When you mean to rename `whitelist`, I derived the name from the gorm doc:
"The general rule to avoid SQL injection is don’t trust user-submitted data, you can perform **whitelist** validation to test user input against an existing set of know"